src/query/common.hpp

```diff
@@ -100,7 +98,7 @@ std::partial_ordering TypedValueCompare(TypedValue const &a, TypedValue const &b
       case TypedValue::Type::Duration:
       case TypedValue::Type::Graph:
       case TypedValue::Type::Function:
-        LOG_FATAL("Invalid type");
+        return a.type() <=> b.type();
```

Comment:
This compared types not values.
Hence does not solve the issue of doing conversion and ordering of different temporal types.